driver/src/contracts/stablex_contract.rs

```diff
@@ -44,6 +47,60 @@ pub trait StableXContract {
     ) -> Result<()>;
 }
 
+struct AuctionElement {
+    valid_from: U256,
+    valid_until: U256,
+    order: Order,
+}
+
+impl AuctionElement {
+    fn from_bytes(bytes: &[u8; 113]) -> Self {
+        let mut data_vector = bytes.to_vec();
+        println!("{}", bytes.len());
+        let account_id = H160::pop_from_log_data(&mut data_vector);
+        // TODO - not sure what this is for.
+        let _sell_token_balance = U256::pop_from_log_data(&mut data_vector);
+        println!("{}", bytes.len());
+        let buy_token = u16::pop_from_log_data(&mut data_vector);
+        let sell_token = u16::pop_from_log_data(&mut data_vector);
+        println!("{}", bytes.len());
+        let valid_from = U256::from(u32::pop_from_log_data(&mut data_vector));
+        let valid_until = U256::from(u32::pop_from_log_data(&mut data_vector));
+        // TODO - not sure about this boolean...
+        u8::pop_from_log_data(&mut data_vector);
+        let is_sell_order = true;
+        println!("{}", bytes.len());
+        let price_numerator = u128::pop_from_log_data(&mut data_vector);
+        let price_denominator = u128::pop_from_log_data(&mut data_vector);
+        let remaining_amount = u128::pop_from_log_data(&mut data_vector);
+
+        // Todo - Will likely have to compute this differently.
+        // TODO - put in own function fn [u128; 3] -> [u128; 2]
+        let buy_amount: u128;
+        let sell_amount: u128;
+        if is_sell_order {
+            sell_amount = remaining_amount;
+            buy_amount = (price_numerator * remaining_amount) / price_denominator;
+        } else {
+            buy_amount = remaining_amount;
+            sell_amount = (price_denominator * remaining_amount) / price_numerator;
+        }
+
+        AuctionElement {
+            valid_from,
+            valid_until,
+            order: Order {
+                batch_information: None,
```

Comment:
We probably need to set the current batch index here?

Reply:
Yes we certainly do!